crates/sui-bridge/src/client/bridge_authority_aggregator.rs

```diff
@@ -163,9 +164,25 @@ async fn request_sign_bridge_action_into_certification(
     clients: Arc<BTreeMap<BridgeAuthorityPublicKeyBytes, Arc<BridgeClient>>>,
     state: GetSigsState,
 ) -> BridgeResult<VerifiedCertifiedBridgeAction> {
-    let (result, _) = quorum_map_then_reduce_with_timeout(
+    // `preferences` is used as a trick here to influence the order of validators to be requested
+    // if `Some(_)`, then we will request validators in the order of the voting power
+    // if `None`, we still refer to voting power, but they are shuffled by randomness
+    // Because ethereum gas price is not negligible, when the signatures are to be verified on ethereum,
+    // we pass in `Some` to make sure the validators with higher voting power are requested first
+    // to save gas cost.
+    let preference = if matches!(action, BridgeAction::SuiToEthBridgeAction(_)) {
+        Some(BTreeSet::new())
+    } else if matches!(action, BridgeAction::EthToSuiBridgeAction(_))
+        || action.chain_id().is_sui_chain()
+    {
+        None
+    } else {
+        Some(BTreeSet::new())
+    };
```

Comment:
I find this condition very difficult to read, and I guess is the `matches` used with `if` or the `||` condition. Not sure.
In any case is this the same as 
```
let preferences = match action {
  BridgeAction::SuiToEthBridgeAction(_) => Some(BTreeSet::new()),
  BridgeAction::EthToSuiBridgeAction(_) => None,
  _ =>  {
    if action.chain_id().is_sui_chain() {
      None
    } else {
      Some(BTreeSet::new())
  }
}

```